shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -2513,6 +2541,14 @@ - (void)takeKeyboardScreenshotAndDisplay {
   _keyboardViewContainer.frame = _keyboardRect;
 }
 
+- (void)showNativeEditMenu:(NSDictionary*)dictionary {
+  if (@available(iOS 16.0, *)) {
+    // TODO: get targetRect from args
+    CGRect targetRect = CGRectMake(100, 100, 100, 100);
```

Comment:
@justinmc I found [an API](https://developer.apple.com/documentation/uikit/uieditmenuinteractiondelegate/3975821-editmenuinteraction?language=objc) that allows iOS to automatically decide the up/down arrow. Do you think you can provide a "target rect" from the framework side? The target rect is the rectangle that presents the menu.

Reply:
Is that like a bounding box for the menu? Currently most menus use an `anchorAbove` and `anchorBelow`. One thing we could do is to create a full-width rectangle whose top and bottom are given by these two anchors. Or if we need something more specific, [here is where](https://github.com/flutter/flutter/blob/c576f0039d75716f04f82c8b58ca208c756b9835/packages/flutter/lib/src/widgets/text_selection_toolbar_anchors.dart#L27) we calculate the anchors. We should have most of the necessary metrics there, we just might need to rework the context menus to take rectangles.

Reply:
It's more like a bounding box for the selected text area (not the menu). Apple just use the rect's top and bottom points for `anchorAbove` and `anchorBelow` for the edit menu. I think the reason apple asks for the rect rather than the 2 points is for potential UI change that may show context menu on left or right.

Reply:
> Sounds like we are probably ok using the full-width solution mentioned above.

Ya a rect of any width will work. 

> we have it [here](https://github.com/flutter/flutter/blob/c576f0039d75716f04f82c8b58ca208c756b9835/packages/flutter/lib/src/widgets/text_selection_toolbar_anchors.dart#L46) and would just need to pipe it through.

If it's easy on your side, I prefer to take the whole rect, so we are feeding the right info into the API. But either way should work.

Reply:
Sounds good, I'll play around with it when I work on the framework side.